walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -132,4 +134,41 @@
 
         return ApiFutures.transform(snapshotFuture, jobFunction, MoreExecutors.directExecutor());
     }
+
+    /** Returns future of all ACTIVE and eligible job posts in database. */
+    public Future<Collection<Job>> fetchAllEligibleJobs(List<String> skills) throws IOException {
+        // Gets all requirements stable id
+        List<String> requirementsList = Requirement.getAllRequirementIds();
+
+        // Gets a list of requirements which the applicant does not have
+        List<String> negateSkills = new ArrayList<>(requirementsList);
+        negateSkills.removeAll(skills);
+
+        final Query activeJobsQuery = FireStoreUtils.getFireStore()
+                .collection(JOB_COLLECTION)
+                .whereEqualTo(JOB_STATUS_FIELD, JobStatus.ACTIVE);
+
+        // Eligible post: post whose requirements do not contain (field is false)
+        // the skills that applicant does not have
+        Query eligiblePostQuery = activeJobsQuery;
+        for (String negateSkill: negateSkills) {
+            String fieldPath = String.format("%s.%s", JOB_REQUIREMENTS_FIELD, negateSkill);
+            eligiblePostQuery = eligiblePostQuery.whereEqualTo(fieldPath, false);
```

Comment:
Would it work if the key is not inside the job?
E.g at first there are 3 possible requirements, maybe next year there are total 5 possible requirements e.g height above 1.6m, weight above 60kg (just example). Then some of the job will not has that key. Will the query consider them as false?

Reply:
that is a problem, i will fix by adding a check to fill all requirements field at constructor